build-release-tools/lib/ManifestGenerator.py

```diff
@@ -119,9 +121,24 @@ def generate_manifest(self):
 class SpecifyDayManifestGenerator(ManifestGenerator):
     def __init__(self, dest, branch, date, builddir, git_credential=None, force=False, jobs=1):
         self._date = date
+        self._jenkins_author = config.gitbit_identity["username"]
         super(SpecifyDayManifestGenerator, self).__init__(dest, branch, builddir, git_credential=git_credential, force=force, jobs=jobs)
 
     def update_repositories_commit(self, repositories):
         for repo in repositories:
             repo_dir = self.directory_for_repo(repo)
-            repo["commit-id"] = self.repo_operator.get_lastest_merge_commit_before_date(repo_dir,self._date)
+            merge_commit = None
+            jenkins_commit = None
+            try:
+                # Get the last merge commit before the date
+                merge_commit = self.repo_operator.get_latest_merge_commit_before_date(repo_dir,self._date)
+                # Get the last commit of Jenkins before the date
+                jenkins_commit = self.repo_operator.get_latest_author_commit_before_date(repo_dir,self._date, self._jenkins_author)
+            except Exception,e:
+                if merge_commit is None:
+                    raise RuntimeError(e)
```

Comment:
```get_latest_merge_commit_before_date``` and ```get_latest_author_commit_before_date``` all can raise exception, so I think try/cathc here could be more meaningful, more than just transfer the exception of previous level.
For example:
```
except Exception,e:
    raise Exception("Faild to update repositories commit: repo {0} \n{1}".format(repo, e))
```
Describe the exception here with words and attach the the exception of previous level for debug.